test/unittest/tests/block/test_block_session_size.cpp

```diff
@@ -0,0 +1,277 @@
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 2) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 5) == 0);
+        validate_and_free_size_list(bms, 5);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_size_alloc", "[block_session_size]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+
+    SECTION("Allocate with null block manager session and no size cache")
+    {
+        std::shared_ptr<MockSession> session_no_bm = MockSession::buildTestMockSession();
+        WT_SIZE *sz;
+
+        REQUIRE(__wti_block_size_alloc(session_no_bm->getWtSessionImpl(), &sz) == 0);
+        validate_and_free_size_block(sz);
+
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+        validate_and_free_size_block(sz);
+    }
+
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+    SECTION("Allocate with fake zero cache size count")
+    {
+        WT_SIZE *sz;
+
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+        // Construct extent cache with one item.
+        bms->sz_cache = sz;
+        bms->sz_cache_cnt = 0;
+
+        WT_SIZE *cached_sz;
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &cached_sz) == 0);
+        REQUIRE(cached_sz == sz);
+        validate_and_free_size_list(bms, 0);
+        validate_and_free_size_block(sz);
+    }
+
+    SECTION("Allocate with one size in cache ")
+    {
+        WT_SIZE *sz;
+
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+        // Construct extent cache with one item with junk next.
+        uint64_t addr = 0xdeadbeef;
+        for (int i = 0; i < sz->depth; i++)
+            sz->next[i + sz->depth] = reinterpret_cast<WT_SIZE *>(addr);
+        bms->sz_cache = sz;
+        bms->sz_cache_cnt = 1;
+
+        WT_SIZE *cached_sz;
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &cached_sz) == 0);
+        REQUIRE(cached_sz == sz);
+        validate_and_free_size_block(sz);
+    }
+
+    SECTION("Allocate with two sizes in cache ")
+    {
+        WT_SIZE *sz;
+        WT_SIZE *sz2;
+
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz) == 0);
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &sz2) == 0);
+
+        // Construct extent cache with two items.
+        sz->next[0] = sz2;
+        bms->sz_cache = sz;
+        bms->sz_cache_cnt = 2;
+
+        WT_SIZE *cached_sz;
+        REQUIRE(__wti_block_size_alloc(session->getWtSessionImpl(), &cached_sz) == 0);
+        REQUIRE(sz == cached_sz);
+        REQUIRE(sz2 != cached_sz);
+        validate_and_free_size_list(bms, 1);
+        validate_and_free_size_block(cached_sz);
```

Comment:
It's not obvious to me why this SECTION calls
        `validate_and_free_size_block(cached_sz);`
whereas the earlier ones call
        `validate_and_free_size_block(sz);`

Reply:
I have changed it to sz instead. good catch! [6e7c584](https://github.com/wiredtiger/wiredtiger/pull/10982/commits/6e7c5842a48870d4129bbb55cf4c714db0ed2089)